### PR TITLE
Netcore: Let build.ps1 create the dotnet new templates

### PR DESCRIPTION
**Pull request by bergmania**
# Details
When executing the `build.ps1` script, it now also creates an `Umbraco.Templates` nuget package. This package can be used to install the templates using `dotnet new -i Umbraco.Templates`.

The templates package contains one package
1. `umbraco`
    - An empty umbraco solution, but with options to add SqlCE (Use `dotnet new umbraco --help` to see how later)
 
The templates are build using the following current files from the `Umbraco.Web.UI.Netcore` project
- `Program.cs`
- `Startup.cs`
- `appsettings.json`
- `appsettings.Development.json`

These files are combined with some static resources, for the templates located in `~/build/templates`

# Tests
To test this you can type `dotnet new -i Umbraco.Templates.9.0.0.nuspec` form the `build.out` folder after the `build.ps1` script is executed.

Then you install a project using `dotnet new umbraco -n {Name}`

To test the generated project, you need to have your NuGet sources to point to the `build.out` folder to find Umbraco in version 9.0.0

### CODE REVIEW

File: build/build.ps1 [hunk 1]
```diff
@@ -326,10 +328,23 @@
     $this.CopyFiles("$src\Umbraco.Web.UI.NetCore\wwwroot\umbraco\js", "*", "$tmp\WebApp\wwwroot\umbraco\js")
     $this.CopyFiles("$src\Umbraco.Web.UI.NetCore\wwwroot\umbraco\lib", "*", "$tmp\WebApp\wwwroot\umbraco\lib")
     $this.CopyFiles("$src\Umbraco.Web.UI.NetCore\wwwroot\umbraco\views", "*", "$tmp\WebApp\wwwroot\umbraco\views")
+
+
+
+    # Prepare templates
+    Write-Host "Copy template files"
+    $this.CopyFiles("$templates", "*", "$tmp\Templates")
```
Comment (elit0451) [inline build/build.ps1:336]: The bin folder gets copied too. We can exclude it by adding a line below ⬇️

File: build/build.ps1 [hunk 2]
```diff
@@ -326,10 +328,23 @@
...
+
+    Write-Host "Copy program.cs and startup.cs for templates"
+    $this.CopyFiles("$src\Umbraco.Web.UI.NetCore", "Program.cs", "$tmp\Templates\UmbracoSolution")
+    $this.CopyFiles("$src\Umbraco.Web.UI.NetCore", "Startup.cs", "$tmp\Templates\UmbracoSolution")
+    $this.CopyFiles("$src\Umbraco.Web.UI.NetCore", "appsettings.json", "$tmp\Templates\UmbracoSolution")
+    $this.CopyFiles("$src\Umbraco.Web.UI.NetCore", "appsettings.Development.json", "$tmp\Templates\UmbracoSolution")
   })
```
Comment (elit0451) [inline build/build.ps1:343]:
```suggestion

  $this.RemoveDirectory("$tmp\Templates\UmbracoSolution\bin")
  })
```

File: src/Umbraco.Tests.Integration/TestServerTest/UmbracoWebApplicationFactory.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@
 using Umbraco.Core.Models.Membership;
 using Umbraco.Core.Services;
 using Umbraco.Web.Common.Security;
-using Umbraco.Web.UI.BackOffice;
+using Umbraco.Web.UI.NetCore;
 
 namespace Umbraco.Tests.Integration.TestServerTest
```
Comment (elit0451) [inline src/Umbraco.Tests.Integration/TestServerTest/UmbracoWebApplicationFactory.cs:25]: This change slipped into this PR 😁

**Comment by elit0451** [comment]
Overall, the changes look good! 

But when I tried installing the 2 templates (**umbraco** and **umbraco -U**), the elements on the login page and the back-office haven't quite rendered _/check picture ⬇️ /_. Seems like we are missing something on the UmbracoCms nuget package. 
Does it happen for you too?

image

----

And also the solution that I get from the **umbraco -U** contains **amd64** and **x86** folders 🤔 

image

File: src/Umbraco.Tests.Integration/TestServerTest/UmbracoWebApplicationFactory.cs (hunk 3 above)
Comment (bergmania) [reply src/Umbraco.Tests.Integration/TestServerTest/UmbracoWebApplicationFactory.cs:25]: This is expected.. The namespace of startup.cs and program.cs was wrong.

**Comment by bergmania** [comment]
Based on the screenshot, it looks like the `umbraco/config/lang` folder is not there. I will investigate, as I was sure I fixed this :)

**Comment by bergmania** [comment]
The `amd64 ` and `x86` folders is expected. These are part of SqlCE and required

File: build/build.ps1 (hunk 1 above)
Comment (bergmania) [reply build/build.ps1:336]: Good catch

**Comment by elit0451** [comment]
Happy with the changes 💪 Merging

**Pull request merged**